ETLParser.py

```diff
@@ -0,0 +1,166 @@
+            line = self.lines[self.pos]
+            if _h2.match(line):
+                break
+            self.pos = self.pos + 1
+
+    def get_next_table(self):
+        # self.pos is now pointing to an h2 line.
+        # Read up to the next h2 and parse table content.
+
+        # Table name line looks something like this: "## 1.1 PERSON".
+        # Parse out the actual table name.
+        table_name = re.match(r'^\#\#\s*\d+\.\d+\s*(?P<name>.*)$',
+                              self.lines[self.pos])
+        table_name = table_name.group('name').lower()
+
+        self.pos = self.pos + 1
+        if self.lines[self.pos].strip() == '':
+            self.pos = self.pos + 1
+
+        # Everything after the h2 and up to the line
+        # 'Field |Required | Data Type | Description | PEDSnet Conventions'
+        # is table content.
+        content = ''
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            self.pos = self.pos + 1
+
+            if re.match(r'^Field\s*\|\s*Required\s*\|', line):
+                # skip the '--- | --- | --- | --- | ---' line as well
+                self.pos = self.pos + 1
+                break
+
+            content = content + line
+
+        self.add_table(table_name)
+
+        # Read field information.
+        # The format is
+        # 'Field |Required | Data Type | Description | PEDSnet Conventions'
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+
+            if _h2.match(line):
+                break
+
+            field = re.match(r'^(?P<name>[^\|]*)\|(?P<req>[^\|]*)\|'
+                             r'(?P<type>[^\|]*)\|(?P<desc>[^\|]*)(\|'
+                             r'(?P<conv>[^\|]*))?\|?$',
+                             line)
+
+            if field:
+                name = field.group('name').lower().strip()
+                req = field.group('req').lower().strip().startswith('yes')
+                data_type = field.group('type').strip()
+                desc = field.group('desc').strip()
+                conv = field.group('conv')
+                if conv:
+                    conv = conv.strip()
+                else:
+                    conv = ''
+
+                self.add_field(table_name, name, req, data_type, desc, conv)
+
+                self.pos = self.pos + 1
+
+            else:
+                break
+
+        # Read the rest of the table contents, whatever comes after
+        # the fields specification table.
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+
+            if _h2.match(line):
+                break
+
+            content = content + line
+            self.pos = self.pos + 1
+
+        self.add_table_content(table_name, content.strip('\n'))
+
+
+def main(path):
+    print Document(path)
+
+if __name__ == '__main__':
+    if len(sys.argv) != 2:
+        sys.exit('Usage: %s %s' % (sys.argv[0], 'full_path_to_markdown_file'))
```

Comment:
To support taking a file or stream, the path to the file should be optional. If not provided, then default to `sys.stdin` as the file-like object.

```python
if len(sys.argv) == 1:
    main(sys.stdin)
    return

path = sys.argv[1]

# This handles checking if the file exists, so there is no need to do
# it explicitly. Also then ensures the file is properly closed once the
# program exits.
with open(path) as f:
    main(f)
```